### PR TITLE
generate function created with lambda and shared pointers

### PR DESCRIPTION
**Pull request by gitbart74859**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: vector-of-shared-ptrs
🏅 Granted 5 XP for @gitbart74859

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#include "vectorFunctions.hpp"
+#include <algorithm>
+#include <iostream>
+
+std::vector<std::shared_ptr<int>> generate(int countNumber) {
+    std::vector<std::shared_ptr<int>> result(countNumber);
+    std::for_each(begin(result), end(result),
+                  [number = 0](auto& pointer) mutable {
+                      pointer = std::make_shared<int>(number++);
+                  });
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:10]: use std::generate. Don't use for_each when other algorithm fit better. Also remember raw loop is also good.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    return result;
+}
+
+void print(std::vector<std::shared_ptr<int>> vec) {
+    for (auto& element : vec) {
+        std::cout << *element << "\n";
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:16]: check if ptr is != nullptr

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    }
+    std::cout << "\n";
+}
+
+void add10(std::vector<std::shared_ptr<int>> vec) {
+    for (auto& element : vec) {
+        if (element == nullptr)
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:23]: if (!element)

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            continue;
+        *element += 10;
+    }
+}
+
+void sub10(int* const element) {
+    if (element == nullptr)
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:30]: if (!element)

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        return;
+    *element -= 10;
+}
+
+void sub10(std::vector<std::shared_ptr<int>> vec) {
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:35]: pass by const reference, avoid copy

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**